### PR TITLE
Adds initial Dockerfile & GHA to publish to Dockerhub

### LINKED ISSUE #11: Add Dockerfile and Docker Hub support
Add a Dockerfile.
Add GitHub action to support pushing the image to DockerHub.

### PR DESCRIPTION
**Pull request by jackdwyer**
Hi gkoh!

Thanks for this work!
I've added a Dockerfile & GHA to publish the image into dockerhub.

Currently its landing in a dockerhub repository associated with myself -> [jackdwyer/bom_exporter](https://hub.docker.com/r/jackdwyer/bom_exporter)

To achieve this I needed to add [repository secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets). 

I've also hardcoded the repository name - I'm guessing it should be updated to `gkoh`.  

Let me know, I'm happy to make any changes to this PR or if you want to use it as the basis to publish yourself. 

Thanks again

### CODE REVIEW

File: .github/workflows/build-push-dockerhub.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - main
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: build image
+        shell: bash
+        run: |
+          docker build -t jackdwyer/bom_exporter:latest .
```
Comment (gkoh) [inline .github/workflows/build-push-dockerhub.yaml]: I've configured a repository variable "DOCKERHUB_TAG" to parameterise this, please update.
It points to 'gkoh0/bom_exporter:latest'.
Whilst I would happily use your DockerHub repository:
* managing secrets would be messy
* people might get suspicious about the github/dockerhub username mismatch

File: .github/workflows/build-push-dockerhub.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: login to dockerhub
+        shell: bash
+        run: |
+          docker login --username ${{ secrets.DOCKERHUB_USERNAME }} --password ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: publish to dockerhub
+        shell: bash
+        run: |
+          docker push jackdwyer/bom_exporter:latest
```
Comment (gkoh) [inline .github/workflows/build-push-dockerhub.yaml]: Here too, DOCKERHUB_TAG.
And also, perhaps consider `docker/build-push-action`?
https://github.com/marketplace/actions/build-and-push-docker-images

File: .github/workflows/build-push-dockerhub.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: build image
+        shell: bash
+        run: |
+          docker build -t jackdwyer/bom_exporter:latest .
+      - name: login to dockerhub
```
Comment (gkoh) [inline .github/workflows/build-push-dockerhub.yaml]: Maybe try using `docker/login-action`?
https://github.com/marketplace/actions/docker-login

**Review by gkoh — COMMENTED** [review]
@jackdwyer Thank you for the PR!
I have made a few minor comments.

It's nice to see someone (other myself) using this project :slightly_smiling_face:

**Comment by jackdwyer** [comment]
Thanks for the feedback!  I've updated the PR to include the requested changes

**Pull request merged**